src/utils/__tests__/tree-utils.spec.js

```diff
@@ -12,6 +21,65 @@ function generateMatrix(numRows, childCount = 1) {
 const translator = { get: d => d };
 
 describe('tree-utils', () => {
+  describe('anyCycle', () => {
+    let nodes;
+    beforeEach(() => {
+      const node3 = { id: '3', children: [] };
+      const node2 = { id: '2', children: [node3] };
+      const node1 = { id: '1', children: [node2] };
+      nodes = [node1, node2, node3];
+    });
+    it('should not detect a cycle in the nodes', () => {
+      const result = anyCycle(nodes);
+      expect(result).to.be.false;
+    });
+    it('should detect a cycle in the nodes', () => {
+      nodes[2].children = [{ id: '1', children: [] }];
+      const result = anyCycle(nodes);
+      expect(result).to.be.true;
+    });
+  });
+
+  describe('fetchPage', () => {
+    const model = { getHyperCubeData: async () => [{ qMatrix: [{}], qArea: { qHeight: 1 } }] };
+    it('should return empty string when fetch correctly', async () => {
+      const result = await fetchPage([], [], model, 1, 1, 1, 10);
+      expect(result).to.equal('');
+    });
+    it('should return max data when limit reached', async () => {
+      const result = await fetchPage([], [], model, 10, 1, 1, 3);
+      expect(result).to.equal('max-data-limit');
+    });
+  });
+
+  describe('getDataMatrix', () => {
```

Comment:
One with empty datapages would be nice

Reply:
Which part would be empty then?

Reply:
Like this https://github.com/qlik-oss/sn-org-chart/pull/111/commits/dd8e280af7442700a736eb7b9817dbf183ba7107 ?

Reply:
I looked at coveralls and one case on line 97 was missed. Looking at the code, I realize it is the opposite. dataPages is never longer than 1, so i is never greater than 0.

Reply:
I'm not even sure that is possible coming back from engine, but added a test for it anyway (since we do have the logic for some reason)